File: src/main/java/br/com/kronos/backend/aplicacao/curso/SubFase.java
package br.com.kronos.backend.aplicacao.curso;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;
import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirTamanho;

import lombok.Builder;
import lombok.Getter;

@Getter
public class SubFase {
    
    private long id;
    private Integer numero;
    private String nome;
    private String sigla;
    private Integer idTipoPeriodo;
    private Long idFase;

    @Builder
	public SubFase(long id, Integer numero, String nome, String sigla, Integer idTipoPeriodo, Long idFase) {

        this.id = id;
        this.numero = exigirNaoNulo(numero, "Número");
        this.nome = exigirTamanho(nome, "Nome", 1, 30);
        this.sigla = exigirTamanho(sigla, "Sigla", 1, 10);  
        this.idTipoPeriodo = exigirNaoNulo(idTipoPeriodo, "Tipo de período"); 
        this.idFase = exigirNaoNulo(idFase, "Fase (Semestre) ");  
	}
}

File: poc-kronos-frontend/src/model/curso/subfase/SubFase.js
class SubFase {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.nome = dados ? dados.nome : null;
    this.sigla = dados ? dados.sigla : null;
    this.numero = dados ? dados.numero : null;
    this.idTipoPeriodo = dados ? dados.idTipoPeriodo : null;
    this.nomeTipoPeriodo = dados ? dados.nomeTipoPeriodo : null;
    this.idFase = dados ? dados.idFase : null;
  }

  static validations(required, maxLength, numeric) {
    return {
      nome: { required, maxLength: maxLength(30) },
      sigla: { required, maxLength: maxLength(10) },
      numero: { required, numeric },
      idTipoPeriodo: { required },
      idFase: { required }
    };
  }

  static validationMessages() {
    return {
      nome: {
        required: "Nome é obrigatório",
        maxLength: "Nome só aceita no máximo 30 caracteres"
      },
      sigla: {
        required: "Sigla é obrigatória",
        maxLength: "Sigla só aceita no máximo 10 caracteres"
      },
      numero: {
        required: "Número é obrigatório",
        numeric: "Somente número"
      },
      idTipoPeriodo: {
        required: "Tipo de Período é obrigatório"
      },
      idFase: {
        required: "Fase é obrigatória"
      }
    };
  }
}
export default SubFase;
